marine/marine.py

```diff
@@ -186,11 +184,12 @@ def validate_fields(
         return bool(self._marine.validate_fields(fields_c_arr, fields_len))
 
     @staticmethod
-    def _parse_output(output: str) -> List[Optional[str]]:
+    def _parse_output(output: List[str], length: int) -> List[Optional[str]]:
         # TODO: this is a bottleneck. Find a better way to provide output from the c code
-        f = StringIO(output)
-        csv_parsed_output = next(csv.reader(f, delimiter="\t", quotechar='"'), [""])
-        return [value or None for value in csv_parsed_output]
+        return (
+            output[i].decode("utf-8") if output[i] is not None else None
```

Comment:
Can `output[i]` actually be `None`? If so, change the typing.
Also, is `output` an actual `list` or is it some `ctypes` array?

Reply:
Yes, `output[i]` can be None, changed return value to be `List[Optional[str]]`.
`output` is not a list but rather `c_char_p` which is more like a `List[bytes]` like you mentioned in the next comment